### PR TITLE
feat: 3761/direct to sign in

### PR DESCRIPTION
**Pull request by cade-exygy**
# Pull Request Template

## Issue Overview
#3761 
When a user is not signed in and they try to apply to a listing, we want to redirect them to the sign in page instead.
This PR addresses #issue

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [x] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Please include a summary of the change and which issue(s) is addressed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- Check if the user is authenticated in the GetApplication component
- If the user is signed in, the 'Apply Online' button directs to the application like normal
- If the user is not signed in, the button redirects to the sign in page including a redirect param that is set to the listing application.

## How Can This Be Tested/Reviewed?
- Set the SHOW_MANDATED_ACCOUNTS env to TRUE

As a logged out user, go to a listing(that is open for application) and click 'Apply Online'
Confirm you are directed to the sign in page.
If you sign in, you should then be directed to the correct application.

As a signed in user, if you click apply online, you will go directly to the application page.

Describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/public/src/components/listing/GetApplication.tsx [hunk 1]
```diff
@@ -44,6 +47,8 @@ export interface ApplicationsProps {
   listingStatus?: string
   /** Whether or not to block submission of test application */
   preview?: boolean
+  /** Whether or not to direct user to sign in page before applying*/
```
Comment (emilyjablonski) [inline sites/public/src/components/listing/GetApplication.tsx:50]: Looks like this isn't used as a heads up

Comment (cade-exygy) [reply sites/public/src/components/listing/GetApplication.tsx:50]: Good catch, will delete

**Comment by emilyjablonski** [comment]
@cade-exygy Would love to see green tests before a full review, lmk if you would like to pair!

**Comment by emilyjablonski** [comment]
This looks good, want to confirm that the two following flows will be captured in future work: (1) redirecting back after hitting create account, (2) redirecting if a user hits the first page of the application directly

**Comment by cade-exygy** [comment]
> This looks good, want to confirm that the two following flows will be captured in future work: (1) redirecting back after hitting create account, (2) redirecting if a user hits the first page of the application directly

Yes that is being handled in https://app.zenhub.com/workspaces/bloom-5dc32d7144bd400001315dac/issues/gh/bloom-housing/bloom/3756

**Review by emilyjablonski — APPROVED** [review]

**Review by ColinBuyck — APPROVED** [review]
LGTM 🐶 🐕 🦮 🐩 🐕‍🦺

**Pull request merged**